pydantic/_internal/_model_construction.py

```diff
@@ -217,7 +219,12 @@ def build_inner_schema(  # noqa: C901
         try:
             default = getattr(cls, ann_name)
         except AttributeError:
-            fields[ann_name] = FieldInfo.from_annotation(ann_type)
+            # if field has no default value and is not in __annotations__ this means that it is
+            # defined in a base class and we can take it from there
+            if ann_name not in annotations:
+                fields[ann_name] = cls.__fields__[ann_name]
+            else:
+                fields[ann_name] = FieldInfo.from_annotation(ann_type)
```

Comment:
```suggestion
            if ann_name in annotations:
                fields[ann_name] = FieldInfo.from_annotation(ann_type)
            else:
                fields[ann_name] = cls.__fields__[ann_name]
```
Generally better to use to test for the "yes" case, not the "not yes" case.